pybop/problems/fitting_problem.py

```diff
@@ -66,12 +66,12 @@ def __init__(
             self._model.n_time_data = self.n_time_data
 
             # Build the model from scratch
-            if self._model._built_model is not None:
-                self._model._model_with_set_params = None
-                self._model._built_model = None
-                self._model._built_initial_soc = None
-                self._model._mesh = None
-                self._model._disc = None
+            if self._model.built_model is not None:
+                self._model.model_with_set_params = None
+                self._model.built_model = None
+                self._model.built_initial_soc = None
+                self._model.mesh = None
+                self._model.disc = None
```

Comment:
Let's replace this with a `self._model.clear()` or `reset()` function defined in BaseModel, so that we can avoid adding public setters.

Reply:
Sounds fine - this should be done in #445 though, as this PR is aimed as ruff linting only, the scope of this PR has been to only make the changes needed for SLF001 to pass.

Reply:
We should try to address the reasons why this standard exists rather than add workarounds, where possible. My request is only a small change.

Reply:
Thanks for the suggestion, I'm going to continue and do this in #445 however. It's a better spot to assess the setters and implement the suggestion above. If this is the only hold up, I'll proceed with merging.